pingcap/tidb
# planner,session: add Reload and File privilege, check Reload for the flush statement

### What problem does this PR solve? 

Fix https://github.com/pingcap/tidb/issues/14213

### What is changed and how it works?

The Flush statements should check the RELOAD privilege. RELOAD is not in our mysql.user table yet.

This PR add RELOAD and FILE columns to the user table, and it checks RELOAD for the flush statements.

### Check List 

Tests 

 - Unit test
 - Manual test (add detailed scripts or steps below)

Manual check the upgrade process and see things like show / grant works right.

Code changes

 - Has persistent data change

Side effects

 - Breaking backward compatibility
Add new columns to `mysql.user`
